### DISCUSSION TITLE
Issue with LR find / fit_one_cycle

### DISCUSSION THREAD

**Discussion by chromade** [timeseriesAI/tsai#74] [General]
Hello, 

Not sure if this an error or - more likely - a mistake by me as I am quite new to fastai/tsai 

Following tutorial 1 using frequency data, lr_find yields a straight line and fit_one_cycle does not work either as shown below. 
I also have the assertion error highlighted by geoHell in the Issues section. 

Thanks!

Full code here:

```
import tsai
from tsai.all import *
print('tsai       :', tsai.__version__)
print('fastai     :', fastai.__version__)
print('fastcore   :', fastcore.__version__)
print('torch      :', torch.__version__)

tsai       : 0.2.15
fastai     : 2.2.7
fastcore   : 1.3.19
torch      : 1.7.1

import pandas as pd
import numpy 
import sys, os 

```

`all_data = pd.read_csv('/home/msms-dl/DbgDir01/data_set_for_time_series_modelling.csv')`

  | direct_num | sample_num | y_list | 0 | 1 | 2 | 3 | 4 | 5 | 6 | ... | 90 | 91 | 92 | 93 | 94 | 95 | 96 | 97 | 98 | 99
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
0 | 0 | 0 | 0.224494 | 0.364835 | 0.520462 | 0.783266 | 0.645734 | 0.512022 | 0.632783 | ... | 0.168986 | 0.156310 | 0.023165 | 0.214470 | 0.265909 | 0.416136 | 0.750068 | 0.491910 | 0.137570 | 0.094487
0 | 0 | 0 | 0.206774 | 0.352226 | 0.482965 | 0.759031 | 0.602404 | 0.457867 | 0.607926 | ... | 0.161466 | 0.157124 | 0.040742 | 0.208043 | 0.285894 | 0.458393 | 0.774518 | 0.486450 | 0.115466 | 0.111570
0 | 0 | 0 | 0.186787 | 0.337114 | 0.440572 | 0.730284 | 0.555950 | 0.402950 | 0.579868 | ... | 0.151666 | 0.157832 | 0.059492 | 0.199419 | 0.303068 | 0.501200 | 0.794325 | 0.479213 | 0.091583 | 0.129218
0 | 0 | 0 | 0.165137 | 0.319422 | 0.394515 | 0.697630 | 0.508092 | 0.349335 | 0.548963 | ... | 0.139652 | 0.158393 | 0.077990 | 0.188847 | 0.317104 | 0.543089 | 0.809362 | 0.470215 | 0.066559 | 0.146691
0 | 0 | 0 | 0.142494 | 0.299314 | 0.345786 | 0.662003 | 0.460838 | 0.298788 | 0.515857 | ... | 0.125158 | 0.159286 | 0.095373 | 0.176511 | 0.327910 | 0.582447 | 0.819847 | 0.459235 | 0.041058 | 0.163729
... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ... | ...
0 | 167 | 0 | 0.322240 | 0.539831 | 0.505063 | 0.167877 | 0.633645 | 0.554848 | 0.353178 | ... | 0.373876 | 0.270555 | 0.167381 | 0.181751 | 0.142647 | 0.125436 | 0.207500 | 0.308182 | 0.373462 | 0.184427
0 | 167 | 0 | 0.315026 | 0.525238 | 0.516878 | 0.162293 | 0.629577 | 0.510102 | 0.360265 | ... | 0.348680 | 0.249225 | 0.190153 | 0.168258 | 0.133828 | 0.112630 | 0.229045 | 0.324270 | 0.380510 | 0.205917
0 | 167 | 0 | 0.301370 | 0.509864 | 0.529479 | 0.179382 | 0.621629 | 0.461042 | 0.366502 | ... | 0.324255 | 0.228143 | 0.210581 | 0.153149 | 0.126947 | 0.102250 | 0.248849 | 0.337763 | 0.383555 | 0.228272
0 | 167 | 0 | 0.281651 | 0.493954 | 0.542676 | 0.216512 | 0.609205 | 0.409628 | 0.370260 | ... | 0.301928 | 0.208219 | 0.228157 | 0.136695 | 0.122028 | 0.096530 | 0.266542 | 0.348133 | 0.382708 | 0.250095
0 | 167 | 0 | 0.255783 | 0.477968 | 0.555399 | 0.267130 | 0.592272 | 0.357991 | 0.369419 | ... | 0.283566 | 0.190531 | 0.242686 | 0.119003 | 0.118727 | 0.097081 | 0.281786 | 0.355272 | 0.377765 | 0.270722

```
X, y = df2xy(all_data, feat_col='sample_num', target_col='y_list', data_cols=all_data.columns[3:103])

X.shape, y.shape

((600, 168, 100), (600,))

```

```splits = get_splits(y, valid_size=.2, stratify=True, random_state=23, shuffle=True)
tfms  = [None, [Categorize()]]
dsets = TSDatasets(X, y, tfms=tfms, splits=splits)
dsets


(#600) [(TSTensor(vars:168, len:100), TensorCategory(0)),(TSTensor(vars:168, len:100), TensorCategory(0)),(TSTensor(vars:168, len:100), TensorCategory(0)),(TSTensor(vars:168, len:100), TensorCategory(0)),(TSTensor(vars:168, len:100), TensorCategory(0)),(TSTensor(vars:168, len:100), TensorCategory(0)),(TSTensor(vars:168, len:100), TensorCategory(0)),(TSTensor(vars:168, len:100), TensorCategory(0)),(TSTensor(vars:168, len:100), TensorCategory(0)),(TSTensor(vars:168, len:100), TensorCategory(0))...]

```

```dls = TSDataLoaders.from_dsets(dsets.train, dsets.valid, bs=[64], batch_tfms=[TSStandardize()], num_workers=0)
model = InceptionTime(dls.vars, dls.c)
learn = Learner(dls, model, metrics=accuracy)
```
`learn.lr_find(suggestions=False)`
![index](https://user-images.githubusercontent.com/80253737/110309718-fb0a2200-8001-11eb-98b9-4e343a0dd9bc.png)

`learn.fit_one_cycle(25, lr_max=1e-3)`

epoch | train_loss | valid_loss | accuracy | time
-- | -- | -- | -- | --
0 | 0.000000 | 0.000000 | 1.000000 | 00:00
1 | 0.000000 | 0.000000 | 1.000000 | 00:00
2 | 0.000000 | 0.000000 | 1.000000 | 00:00
3 | 0.000000 | 0.000000 | 1.000000 | 00:00
4 | 0.000000 | 0.000000 | 1.000000 | 00:00
5 | 0.000000 | 0.000000 | 1.000000 | 00:00
6 | 0.000000 | 0.000000 | 1.000000 | 00:00
7 | 0.000000 | 0.000000 | 1.000000 | 00:00
8 | 0.000000 | 0.000000 | 1.000000 | 00:00
9 | 0.000000 | 0.000000 | 1.000000 | 00:00
10 | 0.000000 | 0.000000 | 1.000000 | 00:00
11 | 0.000000 | 0.000000 | 1.000000 | 00:00
12 | 0.000000 | 0.000000 | 1.000000 | 00:00
13 | 0.000000 | 0.000000 | 1.000000 | 00:00
14 | 0.000000 | 0.000000 | 1.000000 | 00:00
15 | 0.000000 | 0.000000 | 1.000000 | 00:00
16 | 0.000000 | 0.000000 | 1.000000 | 00:00
17 | 0.000000 | 0.000000 | 1.000000 | 00:00
18 | 0.000000 | 0.000000 | 1.000000 | 00:00
19 | 0.000000 | 0.000000 | 1.000000 | 00:00
20 | 0.000000 | 0.000000 | 1.000000 | 00:00
21 | 0.000000 | 0.000000 | 1.000000 | 00:00
22 | 0.000000 | 0.000000 | 1.000000 | 00:00
23 | 0.000000 | 0.000000 | 1.000000 | 00:00
24 | 0.000000 | 0.000000 | 1.000000 | 00:00

**Comment by oguiza** [comment]
Hi @chromade, 
Could you please indicate how you create the splits? I can’t see that in the code.

**Reply by chromade** [reply]
Thank you for the response! 

oops, I define splits between df2xy() and tfrms =  as:


`splits = get_splits(y, valid_size=.2, stratify=True, random_state=23, shuffle=True)
splits`

![index](https://user-images.githubusercontent.com/80253737/110338545-69151000-8027-11eb-94b9-cb01b7e7a6b8.png)

```

((#480) [566,413,338,482,422,484,238,388,318,493...],
 (#120) [142,517,533,513,12,42,249,306,247,360...])
```

**Comment by oguiza** [comment]
Thanks @chromade. 
It strage that the train loss and valid loss are both 0, and accuracy is 100%. 
It looks to me that the issue is in the label. All labels seem to be set to 0, so there's nothing to learn. 
The y_list data is of type float. this means this is a regression problem, instead of a classification one. For regression problems you need to use this: 
```
ts = [None, TSRegression()]
```
instead of 
```
ts = [None, Categorize()] # or [None, TSClassification()] they are both the same.
```
Please, try it and let me know if it works.

**Reply by chromade** [reply]
Hi again,

So the table above is copied wrong so[ I have uploaded the notebook to github](https://github.com/chromade/Issue). Here you can see the 3 categories im trying to classify (with 100 variables over 150 timepoints).
I followed your very clear tutorial again but the _dls_ still only recognises 1 label when there are 3 (which Ive highlighted in the above attempt).

Thanks again for your input.

**Reply by oguiza** [reply]
Could you please run this:
```
check_data(X, y, splits)
```
and share the results?
I still think that for some reason all your y's are set to 0.

**Reply by chromade** [reply]
Hi again,

You're right -- all the y's are set to zero and `check_data(X, y, splits)` has the error:

```
X      - shape: [672 samples x 100 features x 150 timesteps]  type: ndarray  dtype:float64  isnan: 0

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-36-5c89bd81473c> in <module>
----> 1 test = check_data(X, y, splits)
      2 test

~/miniconda3/envs/timeseriesAI/lib/python3.8/site-packages/tsai/data/external.py in check_data(X, y, splits, show_plot)
    217         if isinstance(y[0], (Integral, str)):
    218             n_classes = f'{len(np.unique(y))} ({len(y)//len(np.unique(y))} samples per class) {L(np.unique(y).tolist())}'
--> 219             y_is_nan = 'nan' in [c.lower() for c in np.unique(y)]
    220             print(f'y      - shape: {y.shape}  type: {cls_name(y)}  dtype:{y.dtype}  n_classes: {n_classes}  isnan: {y_is_nan}')
    221         else:

~/miniconda3/envs/timeseriesAI/lib/python3.8/site-packages/tsai/data/external.py in <listcomp>(.0)
    217         if isinstance(y[0], (Integral, str)):
    218             n_classes = f'{len(np.unique(y))} ({len(y)//len(np.unique(y))} samples per class) {L(np.unique(y).tolist())}'
--> 219             y_is_nan = 'nan' in [c.lower() for c in np.unique(y)]
    220             print(f'y      - shape: {y.shape}  type: {cls_name(y)}  dtype:{y.dtype}  n_classes: {n_classes}  isnan: {y_is_nan}')
    221         else:

AttributeError: 'numpy.int64' object has no attribute 'lower'

```
I also tried converting the y_list in dataframe from np.int64 into int and then reconstructing the dataframe. But the y coming out of get_splits is still all 0's.

Thanks

**Reply by oguiza** [reply]
Hi @chromade, I've released an updated df2xy in the GitHub repo. It'd be good if you try it again.
